python/pyspark/sql/types.py

```diff
@@ -629,7 +629,7 @@ def jsonValue(self):
 
     @classmethod
     def fromJson(cls, json):
-        pyUDT = json["pyClass"]
+        pyUDT = str(json["pyClass"])
```

Comment:
Tried again and saw the same error without `str(...)`. This is the same issue as reported in https://bugs.python.org/issue21720.